pull_request_template.md

```diff
@@ -8,6 +8,7 @@
 
 #### Checklist (if applicable)
 
+- [ ] PR title contains correct Jira issue key
 - [ ] README update
```

Comment:
🤔 If you are going to do this then I could suggest that we make the checklist mandatory and add some additions to cover all the scenarios:

```suggestion
#### Checklist

Select all that apply to this PR

- [ ] All commit messages contain the correct Jira issue key
- [ ] Commits will not be squashed before merge
- [ ] Commits will be manually squashed before merge
- [ ] Commits will be automatically squashed before merge (Squash and Merge button)
   - [ ] PR title contains correct descriptions and Jira issue key
- [ ] README updated
```

Reply:
I have a feeling it might negatively impact the cost vs. benefit here. 
I feel it's a lot of clicks if we expand the list with 6 more items for a temporary measure. This adds overhead and increases the risk of people either removing the items, or clicking without thinking.

Let's take this to the team and see what others think.

Reply:
The reason I added them is these are all the questions that I need answered to determine what review I need to leave.

Eg if they have 10 individual commits for a single issue and IMO there should be only 1. I wouldn't feel comfortable to approve the PR without knowing if / how these commits are going to be squashed.

So I have to add a comment asking them to "squash all of your commits". But if they were planning to squash in GH this comment isn't valid, but a comment about the PR title might be valid if they are planning to do that.

This just adds another step in the feedback loop that this checklist would short circuit.

Reply:
@iaindurham Fair point. If we want to support both Merge and Squash, we should have a way of communicating this between PR author and reviewer. Can you think of a way to get the same level of visibility and assurance with fewer clicks / less verbosity?

Reply:
@iaindurham What about this? 
The author can state their intention, and the reviewer knows what to check with just one click.

```suggestion
#### Checklist (if applicable)

- [ ] I will squash: PR title conforms to standard commit message format (`feat: XX-1234 ...`)
- [ ] I will merge without squashing: All commits conforms to standard commit message format (`feat: XX-1234 ...`)
- [ ] README update
```

Reply:
@sam-doomguy This is a good option, nice and clear, less clicks